Build failure: dusklight-1.3.1

### Nixpkgs version

- Unstable (26.11)

### Steps to reproduce

Failure when building

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/341722194

### Relevant log output

```console
error: Cannot build '/nix/store/vciqvx75rswprkyl20jswq7r6pa3jf5p-dusklight-1.3.1.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/nbscr07f7dmklijzxiwsl5qsay1k169v-dusklight-1.3.1
       Last 25 log lines:
       >                  from /build/source/extern/aurora/lib/dolphin/card.cpp:1:
       > /build/source/extern/aurora/include/dolphin/os.h:306:9: note: this is the location of the previous definition
       >   306 | #define ASSERT(cond) ASSERTLINE(__LINE__, cond)
       >       |         ^~~~~~
       > [361/1780] Building CXX object extern/aurora/CMakeFiles/aurora_pad.dir/lib/dolphin/pad/pad.cpp.o
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:468:27: warning: multi-character character constant [-Wmultichar]
       >   468 |   if (magic != SBIG('CTRL')) {
       >       |                           ^
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:468:27: warning: multi-character character constant [-Wmultichar]
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:468:27: warning: multi-character character constant [-Wmultichar]
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:468:27: warning: multi-character character constant [-Wmultichar]
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:1158:49: warning: multi-character character constant [-Wmultichar]
       >  1158 | constexpr uint32_t k_keyboardMagic = SBIG('KBND');
       >       |                                                 ^
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:1158:49: warning: multi-character character constant [-Wmultichar]
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:1158:49: warning: multi-character character constant [-Wmultichar]
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:1158:49: warning: multi-character character constant [-Wmultichar]
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:1277:43: warning: multi-character character constant [-Wmultichar]
       >  1277 |     constexpr uint32_t magic = SBIG('CTRL');
       >       |                                           ^
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:1277:43: warning: multi-character character constant [-Wmultichar]
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:1277:43: warning: multi-character character constant [-Wmultichar]
       > /build/source/extern/aurora/lib/dolphin/pad/pad.cpp:1277:43: warning: multi-character character constant [-Wmultichar]
       > [362/1780] Building C object extern/aurora/extern/CMakeFiles/sqlite3.dir/nix/store/w048fdidps36q1fx3ji7sjayq37bvnni-source/sqlite3.c.o
       > ninja: build stopped: subcommand failed.
```

### Additional context

_No response_

### System metadata

- system: `"x86_64-linux"`
 - host os: `Linux 7.1.5-zen1, NixOS, 26.11 (Zokor), 26.11.20260810.2fcb964`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.34.8`
 - nixpkgs: `/nix/store/v3q54nkmkyvlrqihq7gjqpmsgan83049-source`

### Notify maintainers

@liberodark 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.
- [x] I assert that I have read the [automation/AI policy](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy) and that this issue report complies with it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/du/dusklight/package.nix
+++ b/pkgs/by-name/du/dusklight/package.nix
@@ -41,18 +41,18 @@ let
   auroraSrc = fetchFromGitHub {
     owner = "encounter";
     repo = "aurora";
-    rev = "cb2c340d6cde6827387f14c31ce19e5f28a40e09";
-    hash = "sha256-fiAe5DChCFeakI3pga/g0tXd27osnhQtMBQchuP2NwQ=";
+    rev = "22351fb0b76a4f4f2c4a4dff95aa300101e861aa";
+    hash = "sha256-gDY09IKlTu1Pj7mKw4eQYVXFDJ7j/aTp0iT/Ke93s6s=";
   };
   dawnSrc = fetchzip (
     {
       x86_64-linux = {
-        url = "https://github.com/encounter/dawn-build/releases/download/v20260523.201736/dawn-linux-x86_64.tar.gz";
-        hash = "sha256-KkdlSeiaw2gbQa+phZOpgbequshxQaFITzFdiuGBZvc=";
+        url = "https://github.com/encounter/dawn-build/releases/download/v20260603.191052/dawn-linux-x86_64.tar.gz";
+        hash = "sha256-yTanM4TUIv6akgpt2tai/2W6q4RAt48CxKobRgxK8WU=";
       };
       aarch64-linux = {
-        url = "https://github.com/encounter/dawn-build/releases/download/v20260523.201736/dawn-linux-aarch64.tar.gz";
-        hash = "sha256-accDTIBzgByQ8Rk2a1dAm85s8hj9SYI7NoHkih0vvAg=";
+        url = "https://github.com/encounter/dawn-build/releases/download/v20260603.191052/dawn-linux-aarch64.tar.gz";
+        hash = "sha256-C9pMaUOkSknwQfIw45+WIFve+noTiqud+RiQ0/kV4fA=";
       };
     }
     .${stdenv.hostPlatform.system}
@@ -77,13 +77,13 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "dusklight";
-  version = "1.3.1";
+  version = "1.4.1";
 
   src = fetchFromGitHub {
     owner = "TwilitRealm";
     repo = "dusklight";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-TarPuCE6bjn4nGtHU54pE2gAo/dIgTyghWn7hjdFFgk=";
+    hash = "sha256-wrbsgK9lAiO3+qfQcQ/o+6zEFS27CymJm76FAEanrdM=";
   };
 
   strictDeps = true;
@@ -154,6 +154,8 @@ stdenv.mkDerivation (finalAttrs: {
     (lib.cmakeFeature "AURORA_SDL3_PROVIDER" "system")
     (lib.cmakeFeature "AURORA_NOD_PROVIDER" "system")
     (lib.cmakeFeature "AURORA_NOD_LINKAGE" "static")
+    (lib.cmakeFeature "CMAKE_CXX_FLAGS_INIT" "-include cstring")
+    (lib.cmakeBool "BUILD_SHARED_LIBS" false)
     (lib.cmakeBool "CMAKE_CROSSCOMPILING" true)
     (lib.cmakeBool "DUSK_ENABLE_SENTRY_NATIVE" false)
   ];

